spark-cluster/docker/base/Dockerfile

```diff
@@ -1,11 +1,11 @@
 FROM adoptopenjdk/openjdk8
```

Comment:
Let's try updating the base Docker image to a JDK 17 now that Spark 3.5.0 supports it. [This image](https://hub.docker.com/layers/library/openjdk/17/images/sha256-98f0304b3a3b7c12ce641177a99d1f3be56f532473a528fda38d53d519cafb13?context=explore) seems like a good candidate.

Give your 👍 once the `build.sh` script works well.